packages/use-intl/src/useTranslations.tsx

```diff
@@ -27,19 +30,12 @@ export default function useTranslations<
     TargetKey extends MessageKeys<
       NestedValueOf<
         {'!': IntlMessages},
-        NamespaceKeys<IntlMessages, NestedKeyOf<IntlMessages>> extends NestedKey
-          ? '!'
-          : `!.${NestedKey}`
+        [NestedKey] extends [never] ? '!' : `!.${NestedKey}`
```

Comment:
From [your issue comment](https://github.com/amannn/next-intl/issues/112#issuecomment-1139945855):

> When using generics as types for optional parameters, and the parameter is not given, the generic is defaulted to its constraint.
So I solved this (in a terrible way I did not test) by testing if the constraint extends NestedKey . Obviously, now that I think about it, this will break for objects with only one top-level property...
I also just learnt this very recently, so now I can solve this problem. I've given NestedKey a default "value" (it's actually a type, of course, but at the type-level I refer to types as "values") of never.
And in the conditional, I just check if NestedKey is never: [NestedKey] extends [never] ? ... : .... It seems to work fine.

I think I understand. Ultimately we want to check if `namespace` was provided at the type-level and `never` would mean that it wasn't. Makes sense to me! It's still a bit strange to me though why have to wrap the types into arrays for comparing them with `extends`.

But it's really great to see that this works! 👏